### PR TITLE
Adding in the evaluate postfix notation using Stack

### PR DESCRIPTION
**Pull request by akashgkrishnan**
### **Describe your change:**

Adding in the evaluate postfix notation using Stacks
one of the common use with simple stack question
creating a new file for the data structure of stacks

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by poyea** [comment]
Please change the PR's title to something more meaningful.

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 1]
```diff
@@ -1,12 +1,24 @@
-'''
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py]: This is called a docstring which explains the intent of this file.  The docstring is used by [pydoc](https://docs.python.org/3/library/pydoc.html) if you run the command `python3 -m pydoc data_structures/stacks/evaluate_postfix_notations.py`

By converting this to a comment, this PR would remove that documentation.

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+# Valid operators are +, -, *, /. Each operand may be an integer or another expression.
 
 
 def evaluate_postfix(postfix_notation):
+    """
+    >>> array = ["10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"]
+    >>> evaluate_postfix(array)
+    22
+    >>> array = ["2", "1", "+", "3", "*"]
+    >>> evaluate_postfix(array)
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py:13]:
```suggestion
    >>> evaluate_postfix(["2", "1", "+", "3", "*"])
```
Variables that go out of scope on the very next line should only be used if:
1. they shorten very long lines of code, or
2. their variable name helps to clarify something that is not obvious.

`array` does neither.

Repeat below...

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+    9
+    >>> array = ["4", "13", "5", "/", "+"]
+    >>> evaluate_postfix(array)
+    6
+    >>> array = []
+    >>> evaluate_postfix(array)
+    0
+    """
     operations = {'+', '-', '*', '/'}
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py:22]:
```suggestion
    if not stack:
        return 0
    operations = {'+', '-', '*', '/'}
```
This is a fast-fail.

**Comment by cclauss** [comment]
https://docs.python.org/3/library/operator.html makes dealing with these probems much easier.

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# The Reverse Polish Nation also known as Polish postfix notation or simply postfix notation
+# https://en.wikipedia.org/wiki/Reverse_Polish_notation
+# Classic examples of simple stack implementations
+# Valid operators are +, -, *, /. Each operand may be an integer or another expression.
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py:4]: This is called a docstring which explains the intent of this file.  The docstring is used by [pydoc](https://docs.python.org/3/library/pydoc.html) if you run the command `python3 -m pydoc data_structures/stacks/evaluate_postfix_notations.py`

By converting this to a comment, this PR would remove that documentation.

**Comment by akashgkrishnan** [comment]
thanks, @cclauss for the detailed review. However, pre commits are still failing. I'll update and recommit

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
+"""
+The Reverse Polish Nation also known as Polish postfix notation
+or simply postfix notation. 
+https://en.wikipedia.org/wiki/Reverse_Polish_notation
+Classic examples of simple stack implementations
+Valid operators are +, -, *, /.
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py:6]:
```suggestion
or simply postfix notation.
https://en.wikipedia.org/wiki/Reverse_Polish_notation
Classic examples of simple stack implementations
Valid operators are +, -, *, /.
```

Comment (cclauss) [reply data_structures/stacks/evaluate_postfix_notations.py:6]: Remove trailing whitespace.

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+The Reverse Polish Nation also known as Polish postfix notation
+or simply postfix notation.
+https://en.wikipedia.org/wiki/Reverse_Polish_notation
+Classic examples of simple stack implementations
+Valid operators are +, -, *, /.
+Each operand may be an integer or another expression.
+"""
+
+
+def evaluate_postfix(postfix_notation):
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py:11]:
```suggestion
def evaluate_postfix(postfix_notation: list) -> int:
```
Python type hints

File: data_structures/stacks/evaluate_postfix_notations.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    >>> evaluate_postfix(["2", "1", "+", "3", "*"])
+    9
+    >>> evaluate_postfix(["4", "13", "5", "/", "+"])
+    6
+    >>> array = []
+    >>> evaluate_postfix(array)
```
Comment (cclauss) [inline data_structures/stacks/evaluate_postfix_notations.py:18]:
```suggestion
    >>> evaluate_postfix([])
```

**Comment by akashgkrishnan** [comment]
@cclauss sir is there something else that needs to be fixed in this?

**Review by cclauss — APPROVED** [review]

**Pull request merged**